python/pyarrow/pandas_compat.py

```diff
@@ -318,8 +318,12 @@ def _index_level_name(index, i, column_names):
 
 def dataframe_to_arrays(df, schema, preserve_index, nthreads=1, columns=None,
                         safe=True):
-    if columns is None:
-        columns = df.columns
+    cols = df.columns if schema is None else schema.names
+    if columns is not None:
```

Comment:
IMO `columns` argument should be removed in favor of `schema`.

Reply:
I think it's OK to have both. But they should be mutually exclusive. If you remove columns then you are getting into type inference / conversion issues writing down an exact schema

Reply:
@wesm it breaks backward compatibility here:

https://github.com/apache/arrow/blob/d1185fe98ea74c8886c9d4a549e1684bfefb7e72/python/pyarrow/tests/test_convert_pandas.py#L2099-L2113